website/content/docs/lambda/registration.mdx

```diff
@@ -0,0 +1,279 @@
+
+
+### Optional: Set up a Terminating Gateway
+
+If you intend to invoke Lambda services through a terminating gateway, the gateway must be registered and running in the Consul datacenter. Refer to the following documentation and tutorials for instructions on how to set up a terminating gateway:
+
+*  [Terminating gateways documentation](/docs/connect/gateways#terminating-gateways)
+*  [Terminating gateways on Kubernetes documentation](/docs/k8s/connect/terminating-gateways)
+*  [Connect External Services to Consul With Terminating Gateways tutorial](https://learn.hashicorp.com/tutorials/consul/terminating-gateways-connect-external-services)
+
+To register a Lambda service with a terminating gateway, add the service to the
+`Services` field of the terminating gateway's `terminating-gateway`
+configuration entry. Refer to [Invoke Lambda Functions](/docs/lambda/invocation) for
+more instructions on how to invoke Lambda functions.
+
+### Optional: Run a Mesh Gateway
+
+You can set up a mesh gateway so that you can invoke Lambda services across datacenters and admin partitions. The mesh gateway must be running and registered in the relevant Consul datacenters and partitions. Refer to the following documentation and tutorials for instructions on how to set up mesh gateways:
+
+* [Mesh gateway documentation](/docs/connect/gateways#mesh-gateways)
+* [Connect Services Across Datacenters with Mesh Gateways tutorial](https://learn.hashicorp.com/tutorials/consul/service-mesh-gateways)
+* [Secure Service Mesh Communication Across Kubernetes Clusters tutorial](https://learn.hashicorp.com/tutorials/consul/kubernetes-mesh-gateways?in=consul/kubernetes)
+
+When using admin partitions, you must add Lambda services to the `Services`
+field of [the `exported-services` configuration
+entry](/docs/connect/config-entries/exported-services).
+
+## Automatic Lambda Function Registration
+
+You can deploy the Lambda registrator to your environment to automatically register and deregister Lambda functions with Consul based on the function's tags (refer to the [AWS Lambda tags documentation](https://docs.aws.amazon.com/lambda/latest/dg/configuration-tags.html) to learn about tags).
+The registrator runs as a Lambda function that is invoked by AWS EventBridge (refer to the [AWS EventBridge documentation](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-what-is.html) for additional information).
+EventBridge invokes the registrator using either [AWS CloudTrail](https://docs.aws.amazon.com/lambda/latest/dg/logging-using-cloudtrail.html) to syncronize with Consul in real-time or in [scheduled intervals](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-create-rule-schedule.html).
```

Comment:
```suggestion
You can deploy the Lambda registrator to your environment to automatically register and deregister Lambda functions with Consul based on the function's tags. Refer to the [AWS Lambda tags documentation](https://docs.aws.amazon.com/lambda/latest/dg/configuration-tags.html) to learn about tags.

The registrator runs as a Lambda function that is invoked by AWS EventBridge. Refer to the [AWS EventBridge documentation](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-what-is.html) for additional information.

EventBridge invokes the registrator using either [AWS CloudTrail](https://docs.aws.amazon.com/lambda/latest/dg/logging-using-cloudtrail.html) to syncronize with Consul in real-time or in [scheduled intervals](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-create-rule-schedule.html).
```
Is that better? It seemed awkward to me after seeing the parenthetical phrases applied.